testapps/on_device_unit_tests/test_app/main.py

```diff
@@ -1,45 +1,97 @@
-import sys
-if sys.version_info.major < 3:
-    print(('Running under Python {} but these tests '
-           'require Python 3+').format(sys.version_info.major))
-
-import unittest
-import importlib
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
-print('Imported unittest')
+"""
+On device unit test app
+=======================
 
+This is a dynamic test app, which means that depending on the requirements
+supplied at build time, will perform some tests or others. Also, this app will
+have an ui, or not, depending on requirements as well.
 
-class PythonTestMixIn(object):
+For now, we contemplate three possibilities:
+  -  A kivy unittest app (sdl2 bootstrap)
+  -  A unittest app (webview bootstrap)
+  -  A non-gui unittests app
 
-    module_import = None
+If you install/build this app via the `setup_test_app.py` file, an file named
+`app_requirements.txt` will be generated which will contain the requirements
+that we passed to the `setup_test_app.py` via arguments, which will determine
+the unittests that this app will run.
 
-    def test_import_module(self):
-        """Test importing the specified Python module name. This import test
-        is common to all Python modules, it does not test any further
-        functionality.
-        """
-        self.assertIsNotNone(
-            self.module_import,
-            'module_import is not set (was default None)')
+.. note:: This app it's made to be working on desktop and on an android device.
+          Be aware that some of the functionality of this app will only work on
+          an android device. Also should be mentioned that it will run for
+          python2 and python3.
```

Comment:
I know we still support target Python2, but I would keep it a secret by removing that last sentence :stuck_out_tongue: